crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -358,14 +455,44 @@ impl C8YHttpProxyActor {
         c8y_event: C8yCreateEvent,
     ) -> Result<EventId, C8YRestError> {
         let create_event_url = self.end_point.get_url_for_create_event();
+        let req_builder_closure =
+            |token: String, internal_id: String| -> Result<HttpRequestBuilder, C8YRestError> {
+                let updated_c8y_event = update_event_with_new_internal_id(internal_id, &c8y_event);
+                Ok(HttpRequestBuilder::post(&create_event_url)
+                    .header("Accept", "application/json")
+                    .header("Content-Type", "application/json")
+                    .json(&updated_c8y_event)
+                    .bearer_auth(token))
+            };
 
-        let req_builder = HttpRequestBuilder::post(create_event_url)
-            .header("Accept", "application/json")
-            .header("Content-Type", "application/json")
-            .json(&c8y_event);
-        let http_result = self.execute(req_builder).await?;
+        let http_result = self.execute(None, req_builder_closure).await?;
         let http_response = http_result.error_for_status()?;
         let event_response: C8yEventResponse = http_response.json().await?;
         Ok(event_response.id)
     }
 }
+
+fn update_url_with_new_internal_id(internal_id: String, url: &str) -> Result<Url, C8YRestError> {
```

Comment:
Why do we need this function that updates an url already built?

Why not directly call the primary function that build an url given an internal id?

Reply:
As suggested in the previous PR, I still feel that those `C8yEndpoint` functions generating the URLs can be converted to accept the internal id as an `override_id: Option<String>` parameter so that the URLs can be created out of it. If the `override_id` is `None`, just use the once cached in that `C8yEndpoint` itself.

Reply:
refactored, this is not required anymore.